tests/DCPS/Compiler/anonymous_types/test.idl

```diff
@@ -0,0 +1,45 @@
+#include <tao/LongSeq.pidl>
+
+typedef short Short3Array[3];
+typedef short Short2x3Array[2][3];
+typedef double Double2x3x4Array[2][3][4];
+
+typedef sequence<long> LongSequence;
+typedef sequence<long, 3> Long3Sequence;
+typedef string<100> BoundedString;
+
+typedef struct XYZ_ {
+  double x;
+  double y;
+  double z;
+} XYZ;
+
+@topic
+struct TestStruct {
+  char  c;
+  float f;
+  string str;
+  XYZ position;
+
+  Short3Array s3arr;
+  Short2x3Array s2x3arr;
+  Double2x3x4Array d2x3x4arr;
+
+  LongSequence lseq;
+  Long3Sequence l3seq;
+  BoundedString bstr;
+
+  // prefix a_ denotes anonymous type fields
+
+  short a_s3arr[3];
+  short a_s3arr_2nd[3];
+  short a_s2x3arr[2][3];
+  double a_d2x3x4arr[2][3][4];
+
+  sequence<long> a_lseq;
+  sequence<long> a_lseq_2nd;
+  sequence<long, 3> a_l3seq;
+  sequence<long, 3> a_l3seq_2nd;
+  sequence<long, 5> a_l5seq;
+  string<200> a_bstr;
```

Comment:
Would recommend to also add a test case which the anonymous type is used for a IDL keyword and a C++ keyword to make sure all escaping is done correctly